PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiveModeCheck.java

```diff
@@ -0,0 +1,241 @@
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class extends the LocalInspectionTool and is used to inspect the usage of Azure SDK ServiceBus & ServiceBusProcessor clients in the code.
+ * It checks if the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1.
+ * If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1, a problem is registered with the ProblemsHolder.
+ */
+public class ServiceBusReceiveModeCheck extends LocalInspectionTool {
+
+    /**
+     * Build the visitor for the inspection. This visitor will be used to traverse the PSI tree.
+     *
+     * @param holder The holder for the problems found
+     * @return The visitor for the inspection. This is not used anywhere else in the code.
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new ServiceBusReceiveModeVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * This class extends the JavaElementVisitor and is used to visit the Java elements in the code.
+     * It checks for the usage of Azure SDK ServiceBusReceiver & ServiceBusProcessor clients and
+     * whether the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1.
+     * If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1,
+     * a problem is registered with the ProblemsHolder.
+     */
+    static class ServiceBusReceiveModeVisitor extends JavaElementVisitor {
+
+        // Define the holder for the problems found
+        private static ProblemsHolder holder;
+
+        // Define constants for string literals
+        private static final RuleConfig ruleConfig;
+        private static final String SUGGESTION;
+        private static final List<String> METHODS_TO_CHECK;
+        private static final List<String> CLIENTS_TO_CHECK;
+        private static boolean SKIP_WHOLE_RULE;
+
+        private static final Logger LOGGER = Logger.getLogger(ServiceBusReceiveModeCheck.class.getName());
+
+        static {
+            final String ruleName = "ServiceBusReceiveModeCheck";
+            RuleConfigLoader centralRuleConfigLoader = RuleConfigLoader.getInstance();
+
+            // Get the RuleConfig object for the rule
+            ruleConfig = centralRuleConfigLoader.getRuleConfig(ruleName);
+
+            METHODS_TO_CHECK = ruleConfig.getMethodsToCheck();
+            CLIENTS_TO_CHECK = ruleConfig.getClientsToCheck();
+            SUGGESTION = ruleConfig.getAntiPatternMessage();
+            SKIP_WHOLE_RULE = ruleConfig == RuleConfig.EMPTY_RULE || METHODS_TO_CHECK.isEmpty();
+        }
+
+        /**
+         * Constructor for the visitor
+         *
+         * @param holder The holder for the problems found
```

Comment:
i think you'll get a javadoc warning, or at least we usually do about not writing a javadoc for all parameters. In this case `isOnTheFly` is not in this javadoc. You can mention it as unused.